Tables can now be generated from nested data

christophermaier opened the pull request:
Previously, the `table` command could only extract data from top-level
map keys. Now, it can extract data from nested data structures with the
addition of some JSONPath smarts.

Details about the change can be found in the command documentation in
`config.yaml`.

Fixes operable/cog#784

christophermaier committed:
Tables can now be generated from nested data

Previously, the `table` command could only extract data from top-level
map keys. Now, it can extract data from nested data structures with the
addition of some JSONPath smarts.

Details about the change can be found in the command documentation in
`config.yaml`.

Fixes operable/cog#784

vanstee commented on lib/cog_cmd/format/table.rb:
```diff
@@ -36,4 +38,26 @@ def run_command
+      acc << {name: name,
+              path: JsonPath.new("$.#{selector}")}
+    end
+  end
```
Do we support grabbing a key with a dot or equals sign in it? If not, I guess that's ok, but it might be worth calling out in the docs.

christophermaier replied:
Hrmm... I guess not. I'll update docs for now.

christophermaier committed:
Add RSpec tests for table command

This uses an unmerged branch of `cog-rb` that introduces helper code
to manage the lifecycle of a command during unit tests.

christophermaier commented:
Note: as-is, this should not be merged prior to operable/cog-rb#3. When that happens, the Gemfile reference for `cog-rb` in this change should be updated as appropriate.

vanstee commented on Gemfile:
```diff
@@ -1,4 +1,9 @@
+gem 'cog-rb', github: "operable/cog-rb", branch: "cm/rspec"
+gem 'jsonpath', '~> 0.5.8'
+
+group "development" do
```
Guessing we might need to include `"test"` here as well.

vanstee commented on spec/spec_helper.rb:
```diff
@@ -0,0 +1,6 @@
+require 'cog'
+require 'rspec/cog'
```
It does feel a little sneaky to put this file in `rspec/cog` rather than `cog/rspec` but I guess that's what libs like `rspec-rails` do so maybe this is fine.

christophermaier replied:
I think that's the exact one I cribbed from. I don't particularly care much either way, though.

vanstee commented on spec/table_spec.rb:
```diff
@@ -0,0 +1,122 @@
++--------------------+--------+-------+
+eom
+      end
+    end
```
This stuff is pretty cool :+1:
